src/main/java/at/hannibal2/skyhanni/features/dungeon/DungeonFinderFeatures.kt

```diff
@@ -1,189 +1,259 @@
     private var selectedClass = ""
-
-    @SubscribeEvent
-    fun onRenderItemTip(event: RenderItemTipEvent) {
-        if (!LorenzUtils.inSkyBlock || LorenzUtils.skyBlockArea != "Dungeon Hub") return
-        if (!config.floorAsStackSize) return
-
-        val itemName = event.stack.name.removeColor()
-        val invName = InventoryUtils.openInventoryName()
-
-        if (invName == "Select Floor") {
-            if (itemName == "Any") {
-                event.stackTip = "A"
-            } else if (itemName == "Entrance") {
-                event.stackTip = "E"
-            } else if (itemName.startsWith("Floor ")) {
-                event.stackTip = itemName.split(' ').last().romanToDecimalIfNecessary().toString()
-            }
-        } else if (itemName.startsWith("The Catacombs - ") || itemName.startsWith("MM Catacombs -")) {
-            val floor = itemName.split(" - ").last().removeColor()
-            val floorNum = floor.split(' ').last().romanToDecimalIfNecessary().toString()
-            val isMasterMode = itemName.contains("MM ")
-
-            event.stackTip = if (floor.contains("Entrance")) {
-                "E"
-            } else if (isMasterMode) {
-                "M${floorNum}"
-            } else {
-                "F${floorNum}"
-            }
-        } else if (itemName.endsWith("'s Party")) {
-            val floor = event.stack.getLore().find { it.startsWith("§7Floor: ") } ?: return
-            val dungeon = event.stack.getLore().find { it.startsWith("§7Dungeon: ") } ?: return
-            val floorNum = floor.split(' ').last().romanToDecimalIfNecessary().toString()
-            val isMasterMode = dungeon.contains("Master Mode")
-
-            event.stackTip = if (floor.contains("Entrance")) {
-                "E"
-            } else if (isMasterMode) {
-                "M${floorNum}"
-            } else {
-                "F${floorNum}"
-            }
-        }
-    }
+    private var floorStackSize = mutableMapOf<Int, String>()
+    private var highlightParty = mutableMapOf<Int, LorenzColor>()
+    private var toolTipMap = mutableMapOf<Int, MutableList<String>>()
+    private var inInventory = false
 
     @SubscribeEvent
     fun onInventoryOpen(event: InventoryFullyOpenedEvent) {
-        if (!LorenzUtils.inSkyBlock || LorenzUtils.skyBlockArea != "Dungeon Hub") return
-        if (event.inventoryName != "Catacombs Gate") return
+        if (!isEnabled()) return
+        val inventoryName = event.inventoryName
 
-        val lore = event.inventoryItems[45]?.getLore() ?: return
+        DelayedRun.runDelayed(Duration.parse("10ms")) {
+            floorStackSize.clear()
+            highlightParty.clear()
+//
+//        Code for Stack Tip
+//
+            if(catacombsGatePattern.matches(inventoryName)) {
+                val lore = event.inventoryItems[45]?.getLore()
+                inInventory = true
+                if (!lore.isNullOrEmpty() && lore[0] == "§7View and select a dungeon class.") {
```

Comment:
Change to remove IndexOutOfBound
```suggestion
                if (lore.size > 3 && lore[0] == "§7View and select a dungeon class.") {
```